inbox/mailsync/backends/imap/generic.py

```diff
@@ -906,16 +925,19 @@ def refresh_flags_impl(self, crispin_client, max_uids):
         log.debug(
             "Changed flags refresh response, persisting changes", max_uids=max_uids
         )
-        expunged_uids = set(local_uids).difference(flags.keys())
+        expunged_uids = local_uids.difference(flags)
```

Comment:
Same useless `set()` and `keys()` calls creating useless copies. In case of `keys()` dictionary are already iterated by keys so you can just pass them directly in here.